src/f32ext/ln.rs

```diff
@@ -0,0 +1,113 @@
+use super::abs;
+use super::utils;
+use super::utils::FloatComponents;
+/// Floating natural log
+use core::f32;
+use core::u32;
+
+//excessive precision doesn't matter that much.
+#[allow(clippy::excessive_precision)]
+pub(super) fn ln_1to2_series_approximation(x: f32) -> f32 {
+    // idea from https://stackoverflow.com/a/44232045/
+    // modified to not be restricted to int range and only values of x above 1.0.
+    // and got rid of most of the slow conversions,
+    // should work for all positive values of x.
+    // log base 2(E) == 1/ln(2)
+    //let recip_ln2: f32 = f32::consts::LOG2_E;
+
+    //clippy forces me to do this.
+    if abs::abs(x - 1.0_f32) < f32::EPSILON {
+        return 0.0_f32;
+    }
+    let x_less_than_1: bool = x < 1.0;
+    //use fast inv here?
```

Comment:
Is there a particular fast inverse algorithm you have in mind here?

Reply:
The inverse approximation that you had already in the code.  At this point I'm not sure it is accurate enough for`ln`, especially when `ln` is compounded in a `powf`, and I'm not sure if it would even be the source of speed loss in a no std environment.